library/std/src/thread/local.rs

```diff
@@ -499,20 +536,28 @@ pub mod os {
         }
 
         pub unsafe fn get(&'static self, init: fn() -> T) -> Option<&'static T> {
-            let ptr = self.os.get() as *mut Value<T>;
+            // SAFETY: No mutable references are ever handed out meaning getting
+            // the value is ok.
```

Comment:
Is this again a requirement for the caller?

Reply:
I don't think so: `os::Key<T>` has no other implementations I can see and in the one in this file, no mutable pointers or reference are ever handed out.

Reply:
This is definitely a requirement for the caller to uphold, because we cannot *locally* assert that.